### PR TITLE
[Transform] Transform Reset API

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR introduces Transform Reset API.
Analogously to Anomaly Detection Jobs Reset API, Transform Reset API will reset the transform i.e. bring it to the fresh (newly created) state.
Request is of the form:
```
POST _transform/{transform_id}/_reset
```

Note: Since this is `8.1` feature, it does not need to be added to HLRC.

Relates https://github.com/elastic/elasticsearch/issues/75768

### CODE REVIEW

**Review by sethmlarson — APPROVED** [review]
Specs and YAML tests LGTM

**Comment by lcawl** [comment]
I added 0e76108e70f06b82506c90fae10e5f455f055942 so that the new page will start showing up in the Elasticsearch Guide. Docs now LGTM locally.

File: docs/reference/transform/apis/reset-transform.asciidoc [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+== {api-request-title}
+
+`POST _transform/<transform_id>/_reset`
+
+[[reset-transform-prereqs]]
+== {api-prereq-title}
+
+* Requires the `manage_transform` cluster privilege. This privilege is included
+in the `transform_admin` built-in role.
+* Before you can reset the {transform}, you must stop it.
```
Comment (lcawl) [inline docs/reference/transform/apis/reset-transform.asciidoc:23]:
```suggestion
* Before you can reset the {transform}, you must stop it; alternatively, use the `force` query parameter.
```

**Review by lcawl — APPROVED** [review]
One comment, otherwise docs LGTM

**Comment by przemekwitek** [comment]
> I added 0e76108 so that the new page will start showing up in the Elasticsearch Guide. Docs now LGTM locally.

Thank you!

File: docs/reference/transform/apis/reset-transform.asciidoc (hunk 1 above)
Comment (przemekwitek) [reply docs/reference/transform/apis/reset-transform.asciidoc]: Done.

**Review by hendrikmuhs — APPROVED** [review]
LGTM

**Pull request merged**